gm4_vertical_rails/data/gm4_vertical_rails/functions/process.mcfunction

```diff
@@ -0,0 +1,11 @@
+# tag minecarts upon attaching to a Vertical Rail
+execute as @s[tag=!gm4_vert_rail_momentum] at @s if block ~ ~ ~ ladder run function gm4_vertical_rails:rail/check_direction
+
+# while on a Vertical Rail
+execute as @s[tag=gm4_vert_rail_momentum] at @s if block ~ ~ ~ ladder run function gm4_vertical_rails:rail/apply_direction
+
+# once off a Vertical Rail
+execute as @s at @s unless block ~ ~ ~ ladder run function gm4_vertical_rails:rail/clear_tags
```

Comment:
Reduce the amount of reference frame changes; View my comment on `check_direction.mcfunction`.